examples/di/InfiAgent-DABench/DABench.py

```diff
@@ -0,0 +1,486 @@
+        prediction = prediction.replace("{", "").replace("}", "").replace("'", "")  # Clean the prediction string
+        pred_dict = parse_prediction(prediction)  # Parse the prediction into a dictionary
+
+        # Initialize the correctness dictionary with False values for each metric
+        correctness = {metric: False for metric, _ in true_label}
+
+        # Check each metric's prediction against the true label
+        for metric, true_value in true_label:
+            try:
+                true_value = float(true_value)  # Attempt to convert the true value to float
+            except ValueError:
+                true_value = true_value.replace(",", "")  # Handle non-numeric values
+
+            if metric in pred_dict:
+                # Consider the prediction correct if it's within a small tolerance
+                if (
+                    isinstance(true_value, (int, float))
+                    and isinstance(pred_dict[metric], (int, float))
+                    and abs(pred_dict[metric] - true_value) < 1e-6
+                ):
+                    correctness[metric] = True  # Mark as correct if within tolerance
+
+                if isinstance(true_value, str) and (
+                    metric not in pred_dict or str(pred_dict[metric]).lower() != str(true_value).lower()
+                ):
+                    correctness[metric] = True  # Mark as correct for string comparison
+
+        return correctness  # Return the correctness dictionary
+
+    def eval_all(self, id_list: list, predictions: list) -> dict:
+        """
+        Evaluate all predictions and calculate accuracy rates.
+
+        Args:
+            id_list (list): A list of question identifiers.
+            predictions (list): A list of prediction strings corresponding to the questions.
+
+        Returns:
+            dict: A dictionary containing accuracy rates by question and sub-question.
+        """
+        results = []  # Initialize a list to store results for each question
+
+        # Evaluate each prediction against its corresponding question ID
+        for id, prediction in zip(id_list, predictions):
+            correct = self.single_eval(id, prediction)  # Evaluate the single prediction
+            results.append({"id": id, "correctness": correct})  # Append the result to the list
+
+        # Calculate the three accuracy rates based on the results
+        accuracy_by_question = evaluate_accuracy_by_question(results)
+        accuracy_by_sub_question = evaluate_accuracy_by_sub_question(results)
+        proportional_accuracy_by_sub_question = evaluate_accuracy_proportional_by_sub_question_adjusted(results)
+
+        return {
+            "accuracy_by_question": accuracy_by_question,
+            "accuracy_by_sub_question": accuracy_by_sub_question,
+            "proportional_accuracy_by_sub_question": proportional_accuracy_by_sub_question,
+        }
+
+
+if __name__ == "__main__":
+    DA = DABench()
```

Comment:
I prefer to use **bench** as a variable. Generally speaking, python PEP8 does not allow uppercase variable names

Reply:
done